tests/test_storm_control.py

```diff
@@ -0,0 +1,528 @@
+        status, fvs = atbl.get(dvs.asicdb.portnamemap[if_name])
+        assert status == True
+        print ("\nPort field-value pairs")
+        print (fvs)
+
+        policer_oid = 0
+
+        for fv in fvs:
+            if fv[0] == "SAI_PORT_ATTR_BROADCAST_STORM_CONTROL_POLICER_ID":
+                policer_oid = fv[1]
+
+        if policer_oid != 0:
+            atbl = swsscommon.Table(self.adb,"ASIC_STATE:SAI_OBJECT_TYPE_POLICER")
+            status, fvs = atbl.get(policer_oid)
+            assert status == True
+            print ("POlicer parameters")
+            print (fvs)
+
+        bps = 0
+
+        for fv in fvs:
+            if fv[0] == "SAI_POLICER_ATTR_CIR":
+                bps = fv[1]
+
+        #Retrieved value of bps from ASIC_DB is converted back to user input kbps
+        kbps = int(bps) / int(1000) * 8
+        print ("KBPS value")
+        print (kbps)
+
+        assert str(kbps) == str(kbps_value)
+
+    def test_add_uucast_storm(self,dvs,testlog):
+        self.setup_db(dvs)
+
+        if_name = "Ethernet0"
+        storm_type = "unknown-unicast"
+        #User input is Kbps
+        #Orchagent converts the value to CIR as below and programs the ASIC DB
+        #kbps_value * 1000 / 8
+        kbps_value = 1000000
+
+        self.add_storm_session(if_name, storm_type, kbps_value)
+
+        tbl = swsscommon.Table(self.cdb,"PORT_STORM_CONTROL")
+        (status,fvs) = tbl.get(if_name+"|"+storm_type)
+
+        assert status == True
+        assert len(fvs) > 0
+
+        port_oid = dvs.asicdb.portnamemap[if_name]
+
+        atbl = swsscommon.Table(self.adb,"ASIC_STATE:SAI_OBJECT_TYPE_PORT")
+        status, fvs = atbl.get(dvs.asicdb.portnamemap[if_name])
+        assert status == True
+        print ("\nPort field-value pairs")
+        print (fvs)
+
+        policer_oid = 0
+
+        for fv in fvs:
+            if fv[0] == "SAI_PORT_ATTR_FLOOD_STORM_CONTROL_POLICER_ID":
+                policer_oid = fv[1]
+
+        if policer_oid != 0:
+            atbl = swsscommon.Table(self.adb,"ASIC_STATE:SAI_OBJECT_TYPE_POLICER")
+            status, fvs = atbl.get(policer_oid)
+            assert status == True
+            print ("Policer parameters")
+            print (fvs)
+
+        bps = 0
+
+        for fv in fvs:
+            if fv[0] == "SAI_POLICER_ATTR_CIR":
+                bps = fv[1]
+
+        #Retrieved value of bps from ASIC_DB is converted back to user input kbps
+        kbps = int(bps) / int(1000) * 8
+        print ("KBPS value")
+        print (kbps)
+
+        assert str(kbps) == str(kbps_value)
+
+    def test_add_umcast_storm(self,dvs,testlog):
+        self.setup_db(dvs)
+
+        if_name = "Ethernet0"
+        storm_type = "unknown-multicast"
+        #User input is Kbps
+        #Orchagent converts the value to CIR as below and programs the ASIC DB
+        #kbps_value * 1000 / 8
+        kbps_value = 1000000
+
+        self.add_storm_session(if_name, storm_type, kbps_value)
```

Comment:
replace all the code in this method with call to add_storm_control_interface()

Reply:
Done.